src/wallet/walletdb.h

```diff
@@ -59,6 +63,11 @@ class CHDChain
         READWRITE(this->nVersion);
         READWRITE(nExternalChainCounter);
         READWRITE(masterKeyID);
+        if (this->nVersion >= SUPPORT_WATCHONLY_VERSION) {
+            READWRITE(isWatchOnly);
+            if(isWatchOnly)
+                READWRITE(masterPubKey);
```

Comment:
How is it possible to get an extended private key here? if you provided a xpriv instead of xpub, it should have crashed at startup.